openstack_dashboard/cc_web_sso_urls.py

```diff
@@ -1,19 +1,23 @@
-from django.conf.urls import url
 from django.conf import settings
+from django.conf.urls import url
+from django import http as django_http
+
 from openstack_auth import utils
 from openstack_auth import views
-from openstack_dashboard import cc_websso_views
-utils.patch_middleware_get_user()
 
 
-""" We need websso enabled for Chameleon portal to sso users into Horizon """
-if utils.is_websso_enabled():
-    urlpatterns = [
-        url(r"^login/$", cc_websso_views.login, name='login'),
-        url(r"^logout/$", cc_websso_views.logout, name='logout'),
```

Comment:
Without the ability to log out of a Horizon site independently of the portal, we lose much of the utility we gain from sso. A user logging out of horizon may still have work left to do on portal (and in fact may be doing so on another tab that now lost its session), also, logging out of portal entirely creates a barrier when going back to Horizon on either chi@tacc or chi@uc. Local logout is purposeful and enhances the user experience, it might be worth taking a look at restoring the feature in some capacity

Reply:
It should be possible to restore the "only logout locally" functionality with configuration; right now I have set up `WEBSSO_DEFAULT_REDIRECT_LOGOUT` to point to Portal's logout page. If I remove that configuration value, local logout works (though, it doesn't show a custom-tailored page indicating they can log out of Portal too, though I'm not sure this makes sense in the new SSO model anyways, as both Portal and Keystone(s) are themselves client of another central identity gateway.)

Reply:
I can confirm this works! Just tried out it on dev. I'll leave this "default logout" functionality disabled with a comment on how to enable it if we ever want to do that.

Reply:
Blarggg, but if you use the old SSO method, it will make it so you can't really log out, because it takes you to the login page, which automatically tries to log you in :/

Reply:
In the new federated model, it will be possible to log out of Horizon without logging out of the central IdP, which I think is what you're talking about. I can hack in the logout page again, so that users on the old system will not automatically have their portal session logged out if they log out of Horizon.

Reply:
Agree, restoring the logout pages and pointing there when using the old regular portal sso should work. With a couple of minor changes, we may be able to keep the stand-alone logout  page when using keycloak to let users know that they may still be logged in to keycloak or other auth providers they may have used through keycloak like, `https://identity.tacc.cloud/`

Reply:
Although, since Keycloak logout is working as-is, the decision of whether to integrate the logout page in federated auth could certainly be pushed out to a later release

Reply:
Fixed in 807b319 - I copied much of what you had for the original logout page (will give you a co-authored-by on these) and added a new setting to allow operators to say "yes, I support external logout, but users must confirm they want that." If that's set, then instead of automatically logging them out from the IdP (Portal in this case), we show the logout page with a link to logout from the IdP if they desire.